src/main/java/org/jboss/logmanager/handlers/WriterHandler.java

```diff
@@ -175,9 +177,11 @@ private void writeHead(final Writer writer) {
             final Formatter formatter = getFormatter();
             if (formatter != null) {
                 final String head = formatter.getHead(this);
-                if (checkHeadEncoding) {
-                    if (!getCharset().newEncoder().canEncode(head)) {
-                        reportError("Section header cannot be encoded into charset \"" + getCharset().name() + "\"", null,
+                if (!head.isEmpty() && checkHeadEncoding) {
+                    Charset cs = getCharset();
+                    // UTF-8 is always safe since the UTF-16 chars in String(s) are always encodable
+                    if (!StandardCharsets.UTF_8.equals(cs) && cs.newEncoder().canEncode(head)) {
```

Comment:
Could it be worth keeping a constant set of charsets that can encode any string? e.g. `UTF-8`, `UTF-16`, `UTF-16LE`, etc.?

Reply:
yes, but for few chars we risk to perform N comparisons (although fairly predictable - if it is what mostly happen).
I've picked the default but I'm find to add more too